deployer/deployer-services/src/main/java/io/hyscale/deployer/services/handler/impl/V1PodHandler.java

```diff
@@ -372,4 +393,74 @@ public boolean cleanUp() {
 	public int getWeight() {
 	    return ResourceKind.POD.getWeight();
 	}
+	
+	public void watch(ApiClient apiClient, String appName, String serviceName, String namespace)
+			throws IOException, ApiException, HyscaleException {
+		OkHttpClient httpClient = apiClient.getHttpClient().newBuilder().readTimeout(0, TimeUnit.SECONDS).build();
```

Comment:
Use getUpdatedHttpClient defined in the class, we can update the method to take a timeout as a variable

Reply:
Why is the Timeout zero ?

Reply:
prepare the client in a new method and call it in the watcher
prepareClient() ??

Reply:
This has not been resolved yet ? Can you work on this